bumble/host.py

```diff
@@ -135,13 +138,32 @@ def __init__(self, controller_source=None, controller_sink=None):
         self.link_key_provider = None
         self.pairing_io_capability_provider = None  # Classic only
         self.snooper = None
+        self.pending_connection_roles = {}
 
         # Connect to the source and sink if specified
         if controller_source:
             controller_source.set_packet_sink(self)
         if controller_sink:
             self.set_packet_sink(controller_sink)
 
+    def find_connection_by_bd_addr(
```

Comment:
I suggest to move this logic to device instead, and to use the existing incomplete connections in `Device.pending_connections`

Reply:
That's much better than the original one. Thanks!